crates/aiken-lang/src/builder.rs

```diff
@@ -1363,6 +1355,22 @@ pub fn wrap_validator_args(term: Term<Name>, arguments: &[TypedArg]) -> Term<Nam
     term
 }
 
+pub fn wrap_as_multi_validator(spend: Term<Name>, mint: Term<Name>) -> Term<Name> {
+    Term::equals_integer()
+        .apply(Term::integer(0.into()))
+        .apply(constr_index_exposer(Term::var("__second_arg")))
+        .delayed_if_else(
+            mint.apply(Term::var("__first_arg"))
+                .apply(Term::var("__second_arg")),
+            spend.apply(Term::var("__first_arg")).apply(
+                Term::head_list()
+                    .apply(Term::var(CONSTR_FIELDS_EXPOSER).apply(Term::var("__second_arg"))),
+            ),
+        )
+        .lambda("__second_arg")
+        .lambda("__first_arg")
+}
+
```

Comment:
Not sure to understand what's going on here 🤔; this takes two Term that are seemingly lambdas, and applies one or the other by performing an equality check on the constructor index field of the second argument 🤔? 

The bit I am missing is _why_ this check is _sufficient_ to deduce whether we are in an arity-2 or arity-3 context?